motor/providers/k8s/manifest_provider.go

```diff
@@ -197,29 +198,50 @@ func loadManifestFile(manifestFile string) ([]byte, error) {
 		}
 
 		if fi.IsDir() {
-			// NOTE: we are not using filepath.WalkDir since we do not net recursive walking
-			files, err := ioutil.ReadDir(manifestFile)
-			if err != nil {
-				return nil, err
-			}
-			for i := range files {
-				f := files[i]
-				if f.IsDir() {
-					continue
+			filepath.WalkDir(manifestFile, func(path string, d fs.DirEntry, err error) error {
+				if err != nil {
+					return err
 				}
-				filename := path.Join(manifestFile, f.Name())
 
 				// only load yaml files for now
-				ext := filepath.Ext(filename)
-				if ext == ".yaml" || ext == ".yml" {
-					log.Debug().Str("file", filename).Msg("add file to manifest loading")
-					filenames = append(filenames, filename)
-				} else {
-					log.Debug().Str("file", filename).Msg("ignore file")
+				if !d.IsDir() {
+					ext := filepath.Ext(path)
+					if ext != ".yaml" && ext != ".yml" {
+						log.Debug().Str("file", path).Msg("ignore file, no .yaml or .yml ending")
+						return nil
+					}
+					// check whether this is valid k8s yaml
+					f, err := os.Open(path)
```

Comment:
I think you can replace `os.Open` and `io.ReadAll` with `os.ReadFile()`. That should do the same but it's a single line

Reply:
Thanks, learned something new.